pkg/cmd/run/watch/watch.go

```diff
@@ -134,32 +134,59 @@ func watchRun(opts *WatchOptions) error {
 		prNumber = fmt.Sprintf(" #%d", number)
 	}
 
-	if err := opts.IO.EnableVirtualTerminalProcessing(); err == nil {
-		// clear entire screen
-		fmt.Fprintf(out, "\x1b[2J")
-	}
-
 	annotationCache := map[int64][]shared.Annotation{}
 
 	duration, err := time.ParseDuration(fmt.Sprintf("%ds", opts.Interval))
 	if err != nil {
 		return fmt.Errorf("could not parse interval: %w", err)
 	}
 
+	out := &bytes.Buffer{}
+	opts.IO.StartAlternateScreenBuffer()
 	for run.Status != shared.Completed {
-		run, err = renderRun(*opts, client, repo, run, prNumber, annotationCache)
+		// Write to a temporary buffer to reduce total number of fetches
+		run, err = renderRun(out, *opts, client, repo, run, prNumber, annotationCache)
 		if err != nil {
-			return err
+			break
```

Comment:
same question re: `defer`

Reply:
Same [comment](https://github.com/cli/cli/pull/5681#discussion_r921364614) so I'll resolve this one and we can discuss any possible changes there.